src/entities/queries.rs

```diff
@@ -0,0 +1,43 @@
+pub const MUTUALS_FRIENDS_QUERY: &'static str = "WITH friendsA as (
```

Comment:
Why don't something like the below example? I think it's much more readable and cleaner. I don't like `CASE` that much
```sql
SELECT user_1_friends.address FROM
(                                                  
  SELECT address_2 address 
    FROM friendships WHERE address_1 = '0xA'
  UNION                                         
  SELECT address_1 address 
    FROM friendships WHERE address_2 = '0xA'
) AS user_1_friends
INNER JOIN  
(                                                                                         
  SELECT address_2 address 
    FROM friendships WHERE address_1 = '0xC'
  UNION                                            
  SELECT address_1 address 
    FROM friendships WHERE address_2 = '0xC'
) AS user_2_friends 
ON user_1_friends.address = user_2_friends.address;
```

Reply:
The current query is much faster than using a join:
```
QUERY PLAN
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Hash Semi Join  (cost=1.17..2.33 rows=1 width=32) (actual time=0.186..0.340 rows=4 loops=1)
   Hash Cond: ((CASE WHEN ((f_a.address_1)::text = '0xa'::text) THEN f_a.address_2 ELSE f_a.address_1 END)::text = (CASE WHEN ((f_b.address_1)::text = '0xb'::text) THEN f_b.address_2 ELSE f_b.address_1 END)::text)
   ->  Seq Scan on friendships f_a  (cost=0.00..1.15 rows=2 width=64) (actual time=0.016..0.062 rows=5 loops=1)
         Filter: (((address_1)::text = '0xa'::text) OR ((address_2)::text = '0xa'::text))
         Rows Removed by Filter: 5
   ->  Hash  (cost=1.15..1.15 rows=2 width=64) (actual time=0.135..0.156 rows=4 loops=1)
         Buckets: 1024  Batches: 1  Memory Usage: 9kB
         ->  Seq Scan on friendships f_b  (cost=0.00..1.15 rows=2 width=64) (actual time=0.012..0.094 rows=4 loops=1)
               Filter: (((address_1)::text = '0xb'::text) OR ((address_2)::text = '0xb'::text))
               Rows Removed by Filter: 6
 Planning Time: 0.572 ms
 Execution Time: 0.413 ms
(12 rows)

QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------------------------
 Merge Join  (cost=4.58..4.67 rows=2 width=32) (actual time=0.998..1.602 rows=4 loops=1)
   Merge Cond: ((friendships.address_2)::text = (friendships_2.address_2)::text)
   ->  Unique  (cost=2.29..2.30 rows=2 width=32) (actual time=0.246..0.445 rows=5 loops=1)
         ->  Sort  (cost=2.29..2.29 rows=2 width=32) (actual time=0.232..0.317 rows=5 loops=1)
               Sort Key: friendships.address_2
               Sort Method: quicksort  Memory: 25kB
               ->  Append  (cost=0.00..2.28 rows=2 width=32) (actual time=0.032..0.193 rows=5 loops=1)
                     ->  Seq Scan on friendships  (cost=0.00..1.12 rows=1 width=32) (actual time=0.017..0.047 rows=3 loops=1)
                           Filter: ((address_1)::text = '0xa'::text)
                           Rows Removed by Filter: 7
                     ->  Seq Scan on friendships friendships_1  (cost=0.00..1.12 rows=1 width=32) (actual time=0.007..0.032 rows=2 loops=1)
                           Filter: ((address_2)::text = '0xa'::text)
                           Rows Removed by Filter: 8
   ->  Unique  (cost=2.29..2.30 rows=2 width=32) (actual time=0.184..0.437 rows=4 loops=1)
         ->  Sort  (cost=2.29..2.29 rows=2 width=32) (actual time=0.170..0.244 rows=4 loops=1)
               Sort Key: friendships_2.address_2
               Sort Method: quicksort  Memory: 25kB
               ->  Append  (cost=0.00..2.28 rows=2 width=32) (actual time=0.023..0.163 rows=4 loops=1)
                     ->  Seq Scan on friendships friendships_2  (cost=0.00..1.12 rows=1 width=32) (actual time=0.009..0.031 rows=2 loops=1)
                           Filter: ((address_1)::text = '0xb'::text)
                           Rows Removed by Filter: 8
                     ->  Seq Scan on friendships friendships_3  (cost=0.00..1.12 rows=1 width=32) (actual time=0.011..0.035 rows=2 loops=1)
                           Filter: ((address_2)::text = '0xb'::text)
                           Rows Removed by Filter: 8
 Planning Time: 0.330 ms
 Execution Time: 2.633 ms
(26 rows)
```